### PR TITLE
FEATURE: New pages added to user-activity and user-notifications which shows reactions of current user and reactions made on current users posts. user-activity-reactions and user-notifications-reactions

### PR DESCRIPTION
**Pull request by Ahmedgagan**

### CODE REVIEW

**Comment by Ahmedgagan** [comment]
@jjaffeux

File: app/serializers/user_reaction_serializer.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      next unless reaction_user
+
+      return {
+        id: reaction.reaction_value,
+        type: reaction.reaction_type.to_sym,
+        can_undo: reaction_user.can_undo?,
+        avatar_template: reaction_user.avatar_template,
+        created_at: reaction_user.created_at,
+        reaction_user_id: reaction_user.id
+      } if reaction.reaction_users_count
```
Comment (jjaffeux) [inline app/serializers/user_reaction_serializer.rb]: why not adding this condition to the previous next ?

File: assets/javascripts/discourse/components/discourse-reactions-reaction-post.js.es6 [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  emojiUrl(currentUserReaction) {
+    if(!currentUserReaction) {
+      return;
+    }
+    return emojiUrlFor(currentUserReaction);
+  },
+
+  @discourseComputed("post.current_user_reaction.avatar_template", "currentUser")
+  currentUserAvatar(avatarTemplate, currentUser) {
+    currentUser.avatar_template = avatarTemplate;
```
Comment (jjaffeux) [inline assets/javascripts/discourse/components/discourse-reactions-reaction-post.js.es6]: not a big fan of this, computed property with a side effect is a big smell in my book.

can you do this in didReceiveAttrs ?

File: assets/javascripts/discourse/connectors/user-activity-bottom/discourse-reactions-user-activity-reactions.hbs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{{#if siteSettings.discourse_reactions_enabled}}
+<li>
+  {{#link-to 'userActivity.yourReactions'}}
+    {{d-icon 'far-smile'}} {{i18n 'discourse_reactions.reactions_title'}}
```
Comment (jjaffeux) [inline assets/javascripts/discourse/connectors/user-activity-bottom/discourse-reactions-user-activity-reactions.hbs]: have a span around i18n

File: assets/javascripts/discourse/controllers/user-activity-your-reactions.js.es6 [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    if (!this.canLoadMore) {
+      return;
+    }
+    if (this.loading) {
+      return;
+    }
+    this.set("loading", true);
+    const posts = this.model;
+    if (posts && posts.length) {
+      const beforePostId = posts[posts.length - 1].get("current_user_reaction").id;
```
Comment (jjaffeux) [inline assets/javascripts/discourse/controllers/user-activity-your-reactions.js.es6]: get is not going to save you if posts is not found, eg: null.get(...) doesn’t work

File: assets/javascripts/discourse/discourse-reactions-route-map.js.es6 [hunk 5]
```diff
@@ -0,0 +1,6 @@
+export default {
+  resource: "user.userActivity",
+  map() {
+    this.route("yourReactions", { path: "your-reactions" });
```
Comment (jjaffeux) [inline assets/javascripts/discourse/discourse-reactions-route-map.js.es6]: any reason we don't go with "my-reactions", what do we do at other places?

File: assets/javascripts/discourse/models/discourse-reactions-custom-reaction.js.es6 [hunk 6]
```diff
@@ -17,6 +21,29 @@ CustomReaction.reopenClass({
     ).catch(e => {
       bootbox.alert(`${e.jqXHR.status} ${e.errorThrown}`);
     });
+  },
+
+  findYourReactions(name, opts) {
+    opts = opts || {};
+
+    if(!name ) {
```
Comment (jjaffeux) [inline assets/javascripts/discourse/models/discourse-reactions-custom-reaction.js.es6]: make sure prettier is setup please, indenting is wrong here

File: assets/javascripts/discourse/models/discourse-reactions-custom-reaction.js.es6 [hunk 7]
```diff
@@ -17,6 +21,29 @@ CustomReaction.reopenClass({
...
+      return;
+    }
+
+    const data = {};
+
+    if (opts.beforePostId) {
+      data.before_post_id = opts.beforePostId;
+    }
+
+    return ajax(`/discourse-reactions/posts/reactions-given/${name}.json`, { data }).then((posts) => {
```
Comment (jjaffeux) [inline assets/javascripts/discourse/models/discourse-reactions-custom-reaction.js.es6]: can you try to define and use adapter/store ?

Comment (Ahmedgagan) [reply assets/javascripts/discourse/models/discourse-reactions-custom-reaction.js.es6]: I tried to do this, but `this.store` is undefined here. Do you mean to remove this `findYourReactions` function from here and make an adapter instead?

File: assets/javascripts/discourse/controllers/user-activity-my-reactions.js.es6 [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+import CustomReaction from "../models/discourse-reactions-custom-reaction";
+
+export default Controller.extend({
+  canLoadMore: true,
+  loading: false,
+  application: controller(),
+
+  @action
+  loadMore() {
+    if (!this.canLoadMore) {
```
Comment (jjaffeux) [inline assets/javascripts/discourse/controllers/user-activity-my-reactions.js.es6]: can you group canLoadMore and loading check please?

File: assets/javascripts/discourse/routes/user-activity-my-reactions.js.es6 [hunk 9]
```diff
@@ -0,0 +1,26 @@
+import DiscourseRoute from "discourse/routes/discourse";
+import CustomReaction from "../models/discourse-reactions-custom-reaction";
+
+export default DiscourseRoute.extend({
+  userActionType: 17,
+  noContentHelpKey: "discourse_assigns.no_reactions",
+
+  model() {
+    return CustomReaction.findMyReactions(
+      this.modelFor("user").get("username_lower")
```
Comment (jjaffeux) [inline assets/javascripts/discourse/routes/user-activity-my-reactions.js.es6]:
```suggestion
      this.modelFor("user").username_lower
```

Comment (jjaffeux) [reply assets/javascripts/discourse/routes/user-activity-my-reactions.js.es6]: I'm actually not sure here, what does modelFor returns if we don't get a model for the user ?

File: assets/javascripts/discourse/controllers/user-activity-my-reactions.js.es6 [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  loadMore() {
+    if (!this.canLoadMore || this.loading || !this.reactionsUrl) {
+      return;
+    }
+
+    this.set("loading", true);
+    const posts = this.model;
+
+    const beforeReactionUserId = posts.length
+      ? posts[posts.length - 1].get("id")
```
Comment (jjaffeux) [inline assets/javascripts/discourse/controllers/user-activity-my-reactions.js.es6]: what is get achieving here ?

File: spec/requests/custom_reactions_controller_spec.rb [hunk 11]
```diff
@@ -121,21 +152,21 @@
     SiteSetting.post_undo_action_window_mins = 10
     sign_in(user_1)
     put "/discourse-reactions/posts/#{post_1.id}/custom-reactions/thumbsup/toggle.json"
-    expect(DiscourseReactions::Reaction.count).to eq(1)
-    expect(DiscourseReactions::ReactionUser.count).to eq(1)
-
-    put "/discourse-reactions/posts/#{post_1.id}/custom-reactions/thumbsup/toggle.json"
-    expect(DiscourseReactions::Reaction.count).to eq(0)
-    expect(DiscourseReactions::ReactionUser.count).to eq(0)
+    expect(DiscourseReactions::Reaction.count).to eq(2)
```
Comment (jjaffeux) [inline spec/requests/custom_reactions_controller_spec.rb]: for all this count check you should use the expect to change by xxx form, look at what we do in core for this

File: assets/javascripts/discourse/controllers/user-activity-my-reactions.js.es6 (hunk 10 above)
Comment (Ahmedgagan) [reply assets/javascripts/discourse/controllers/user-activity-my-reactions.js.es6]: Oooh actually its a mistake, i had to change the variable name `posts` to `reaction_users`

As here we are getting reaction_users from backend and we want reaction_users_id

**Review by jjaffeux — APPROVED** [review]

**Pull request merged**